webapp/src/dogsList/listWithDogs.tsx

```diff
@@ -41,11 +93,53 @@ const useStyles = makeStyles((theme: Theme) =>
       display: "flex",
       alignItems: "center",
       marginTop: "2vh",
+      flexGrow: 1,
+      alignSelf: "center",
     },
   })
 );
 
+scheme.configureHeader((builder) => {
+  builder
+    .registerConfig("md", {
+      position: "absolute",
+    })
+    .registerConfig("xs", {
+      position: "relative", // won't stick to top when scroll down
+    });
+});
+
+scheme.configureEdgeSidebar((builder) => {
+  builder
+    .create("unique_id", { anchor: "left" })
+    .registerTemporaryConfig("xs", {
+      width: 200, // 'auto' is only valid for temporary variant
+    })
+    .registerPermanentConfig("md", {
+      width: "15%", // px, (%, rem, em is compatible)
+      collapsible: true,
+      collapsedWidth: "6%",
+    });
+});
+//scheme.configureInsetSidebar
+
 export default function ListWithDogs() {
+  const state = store.getState() as State;
+  const [collapsed, setCollapsed] = useState(false);
+  const dogs = useSelector(
+    (state: State) => state.dogs as ILostDogWithPicture[]
+  );
+  const loading = useSelector((state: State) => state.loading);
+  const refreshRequired = useSelector(
+    (state: State) => state.dogsRequireRefresh
+  );
+  const lastPage = useSelector((state: State) => state.dogsLastPage);
+  const [filters, setFilters] = useState({
```

Comment:
i assume those are the same filters as in fetching, type them (probably import type from fetching), you can use useState<type>() to give the type explicitely